### ISSUE TITLE
PreciseDecimal to JSON number ends up with long list of decimals

### ISSUE THREAD

**Issue by willthrom** [davdroman/PreciseDecimal#11]
It seems from JSON to the PreciseDecimal we get the proper value but when writing back to JSON.... lots of decimal are added.

Not sure this is a problem for this library though. It is like the JSONEncoder ends up from PreciseDecimal to FLOAT to JSON-Number

![image](https://user-images.githubusercontent.com/2103849/201091402-8fdc03b1-8b4d-4c72-856f-c6ccdc2b3db0.png)

<img width="1057" alt="Screen Shot 2022-11-10 at 12 06 05" src="https://user-images.githubusercontent.com/2103849/201090914-20abefdc-969a-4ae9-88b9-88e7585bac68.png">

**Comment by davdroman** [comment]
Yes, I'm afraid this is unavoidable because of how `JSONEncoder` is implemented. I'm planning a major release bump that moves away from literal numbers altogether and instead encodes decimals as strings by default. Details here #10.

**Comment by willthrom** [comment]
It is surprising a back end produce a decimal point number and we cannot return back the same decimal value.

in this case backend is in golang, using the library Decimal, which JSONEncoder seems to do fine. Android seems to be able to parse properly this JSON from Decimal and the encode it back to send to the same backend. 

But with Apple, reading breaks in some situations and although with your library now reading works most of the time (if not always), writing back to JSON Number creates those precision issues.

Here I can only see two possible options, or our back end do the truncate always (or rounding to the nearest), or we change FE<->BE to use strings instead, but that is a big change (besides network usage). I need to find some time to do some research because my initial finding have very almost none.   (We are moving to from FE doing maths operations to MOVE everything to the BE, so no more desaligment between platforms)

**Comment by davdroman** [comment]
> Here I can only see two possible options, or our back end do the truncate always (or rounding to the nearest), or we change FE<->BE to use strings instead, but that is a big change

You're correct. Although truncating/rounding is definitely the riskier choice.

##### PS: I totally understand your discontent. It truly is ridiculous having this gap in the language in 2022 when Android solved it years ago.
